yahp/utils/json_schema_helpers.py

```diff
@@ -0,0 +1,88 @@
+import inspect
+from enum import Enum
+from typing import Any, Dict
+
+from yahp.utils import type_helpers
+
+
+def get_registry_json_schema(f_type: type_helpers.HparamsType, registry: Dict[str, Any]):
+    """Convert type into corresponding JSON Schema. As the given name is in the `hparams_registry`,
+    create objects for each possible entry in the registry and treat as union type.
+    """
+    res = {'anyOf': []}
+    for key, value in registry.items():
+        res['anyOf'].append({
+            'type': 'object',
+            'properties': {
+                key: get_type_json_schema(type_helpers.HparamsType(value))
+            },
+            'additionalProperties': False,
+        })
+    return check_for_list_and_optional(f_type, res)
+
+
+def get_type_json_schema(f_type: type_helpers.HparamsType):
+    """Convert type into corresponding JSON Schema.
+    """
+    # Import inside function to resolve circular dependencies
+    from yahp.auto_hparams import ensure_hparams_cls
+
+    res = {}
+
+    # Union
+    if len(f_type.types) > 1:
+        # Add all union types using anyOf
+        res = {'anyOf': []}
+        for union_type in f_type.types:
+            res['anyOf'].append(get_type_json_schema(type_helpers.HparamsType(union_type)))
+    # Primitive Types
+    elif f_type.types[0] is str:
```

Comment:
make sure to check against all possible types in a union.

```suggestion
    elif any(safe_issubclass(x, str) for x in f_type.types):
```

Reply:
Union is processed above, so at this point we know there is only 1 type